src/cypress-api/index.ts

```diff
@@ -45,13 +50,77 @@ const doArchive = async ({
     },
     allSnapshots,
     Object.fromEntries(bufferedArchiveList),
-    // @ts-expect-error will fix when Cypress has its own package
-    { ...chromaticStorybookParams, viewport: { width: 500, height: 500 } }
+    chromaticStorybookParams
   );
 };
 
-export const archiveCypress = async (params: ArchiveParams): Promise<null> => {
-  await doArchive(params);
+// using a single Watcher instance across all tests (for the test run)
+// each time a test completes, we'll save to disk whatever archives are there at that point.
+// This should be safe since the same resource from the same URL should be the same during the entire test run.
+// Cypress doesn't give us a way to share variables between the "before test" and "after test" lifecycle events on the server.
+let watcher: Watcher = null;
+
+let host = '';
+let port = 0;
+
+export const setupNetworkListener = async (): Promise<null> => {
+  try {
+    const { webSocketDebuggerUrl } = await Version({
+      host,
+      port,
+    });
+
+    const cdp = await CDP({
+      target: webSocketDebuggerUrl,
+    });
+
+    if (!watcher) {
+      watcher = new Watcher(cdp);
+      await watcher.watch();
+    }
+  } catch (err) {
+    console.log('err', err);
+  }
 
   return null;
 };
+
+export const saveArchives = (archiveInfo: WriteParams) => {
+  return new Promise((resolve) => {
+    return watcher.idle().then(() => {
+      // the watcher's archives come from the server, everything else (DOM snapshots, test info, etc) comes from the browser
+      return writeArchives({ ...archiveInfo, resourceArchive: watcher.archive }).then(() => {
+        resolve(null);
+      });
+    });
+  });
+};
+
+// We use this lifecycle hook because we need to know what host and port Chrome Devtools Protocol is listening at.
+export const onBeforeBrowserLaunch = (
```

Comment:
Did you consider making this part unnecessary if you are running in electron and passing in the env var?

I guess that'd complicate the DX and not save people much effort.

Reply:
@tmeasday good point. We could remove the need for this lifecycle if we expected everyone to always pass their port as an environment variable (I suppose we could require hostname as well, but I'd assume it's always `127.0.0.1`).

Right now people only need to pass it if they're using Electron. But that is the default, and having a bunch of "in-this-case-you-need-to-add-this" messaging would not be good DX. WDYT?

Reply:
I'm not sure, perhaps a question for DX / @byebrianwong